### PR TITLE
feature/7836 FAQ-Link if Certificate is not available

### PR DESCRIPTION
**Pull request by ArturFriesen**
## Description
Add handling of labId.
Add faq link to alert.

## Link to Jira
https://jira-ibs.wbs.net.sap/browse/EXPOSUREAPP-7836

### CODE REVIEW

**Review by NataliaLemmerth — APPROVED** [review]
UA approved

File: src/xcode/ENA/ENA/Source/Services/HealthCertificate/Models/TestCertificateRequest.swift [hunk 1]
```diff
@@ -55,6 +58,7 @@ class TestCertificateRequest: Codable, Equatable {
 		encryptedDEK = try container.decodeIfPresent(String.self, forKey: .encryptedDEK)
 		encryptedCOSE = try container.decodeIfPresent(String.self, forKey: .encryptedCOSE)
 		requestExecutionFailed = try container.decode(Bool.self, forKey: .requestExecutionFailed)
+		labId = try container.decode(String.self, forKey: .labId)
```
Comment (nickguendling) [inline src/xcode/ENA/ENA/Source/Services/HealthCertificate/Models/TestCertificateRequest.swift:61]:
```suggestion
		labId = try container.decodeIfPresent(String.self, forKey: .labId)
```

Just for downward compatibility, not super important as this class was not released yet, but the developers and testers might appreciate it.

File: src/xcode/ENA/ENA/Source/Services/HealthCertificate/__test__/HealthCertificateServiceTests.swift [hunk 2]
```diff
@@ -1010,4 +1011,32 @@ class HealthCertificateServiceTests: CWATestCase {
...
+			guard case let .failure(error) = result,
+				  case .dgcNotSupportedByLab = error else {
+				XCTFail("Error dgcNotSupportedByLab was expected.")
+				expectation.fulfill()
+				return
+			}
+			expectation.fulfill()
+		}
+
+		waitForExpectations(timeout: .short)
```
Comment (nickguendling) [inline src/xcode/ENA/ENA/Source/Services/HealthCertificate/__test__/HealthCertificateServiceTests.swift:1040]:
```suggestion
		waitForExpectations(timeout: .short)
		
		XCTAssertEqual(service.testCertificateRequests.value.first, certificateRequest)
		XCTAssertTrue(certificateRequest.requestExecutionFailed)
		XCTAssertFalse(certificateRequest.isLoading)
```

Would be nice to check that the request is stored and in the failed state

File: src/xcode/ENA/ENA/Source/Services/HealthCertificate/HealthCertificateService.swift [hunk 3]
```diff
@@ -202,14 +204,24 @@ class HealthCertificateService {
...
 	func executeTestCertificateRequest(
 		_ testCertificateRequest: TestCertificateRequest,
 		retryIfCertificateIsPending: Bool,
 		completion: ((Result<Void, HealthCertificateServiceError.TestCertificateRequestError>) -> Void)? = nil
 	) {
 		Log.info("[HealthCertificateService] Executing test certificate request: \(private: testCertificateRequest)", log: .api)
+
 		testCertificateRequest.isLoading = true
 
+		// If we didn't retrieved a labId for a PRC test result, the lab is not supporting test certificates.
```
Comment (nickguendling) [inline src/xcode/ENA/ENA/Source/Services/HealthCertificate/HealthCertificateService.swift:217]:
```suggestion
		// If we didn't retrieve a labId for a PRC test result, the lab is not supporting test certificates.
```

**Review by nickguendling — CHANGES_REQUESTED** [review]

**Review by freshking — APPROVED** [review]

**Review by nickguendling — APPROVED** [review]

**Review by marcussc — APPROVED** [review]

**Pull request merged**